src/coremods/core_channel/cmd_kick.cpp

```diff
@@ -42,9 +42,14 @@ CmdResult CommandKick::Handle (const std::vector<std::string>& parameters, User
 	else
 		u = ServerInstance->FindNick(parameters[1]);
 
-	if ((!u) || (!c) || (u->registered != REG_ALL))
+	if (!c)
 	{
-		user->WriteNumeric(Numerics::NoSuchNick(c ? parameters[1] : parameters[0]));
+		user->WriteNumeric(ERR_NOSUCHCHANNEL, parameters[0], "No such channel");
```

Comment:
Can you add something like `Numerics::NoSuchNick` but for channels and use that instead?